source/non_matching/fe_values.cc

```diff
@@ -268,6 +268,217 @@ namespace NonMatching
+    Assert(
+      q_collection_1D.size() == fe_collection->size() ||
+        q_collection_1D.size() == 1,
+      ExcMessage(
+        "Size of hp::QCollection<1> must be the same as hp::FECollection or 1."));
+
+    // For each element in fe_collection, create dealii::FEInterfaceValues
+    // objects to use on the non-intersected cells.
+    fe_values_inside_full_quadrature.resize(fe_collection->size());
+    fe_values_outside_full_quadrature.resize(fe_collection->size());
+    for (unsigned int fe_index = 0; fe_index < fe_collection->size();
+         ++fe_index)
+      {
+        const unsigned int mapping_index =
+          mapping_collection->size() > 1 ? fe_index : 0;
+        const unsigned int q_index = q_collection.size() > 1 ? fe_index : 0;
+
+        fe_values_inside_full_quadrature[fe_index].emplace(
+          (*mapping_collection)[mapping_index],
+          (*fe_collection)[fe_index],
+          q_collection[q_index],
+          region_update_flags.inside);
+        fe_values_outside_full_quadrature[fe_index].emplace(
+          (*mapping_collection)[mapping_index],
+          (*fe_collection)[fe_index],
+          q_collection[q_index],
+          region_update_flags.outside);
+      }
+  }
+
+
+
+  template <int dim>
+  template <bool level_dof_access>
+  void
+  FEInterfaceValues<dim>::do_reinit(
+    const TriaIterator<DoFCellAccessor<dim, dim, level_dof_access>> &cell,
+    const unsigned int                                               face_no,
+    const std::function<void(dealii::FEInterfaceValues<dim> &)> &call_reinit)
+  {
+    current_face_location =
+      mesh_classifier->location_to_level_set(cell, face_no);
+    active_fe_index = cell->active_fe_index();
+
+    // These objects were created with a quadrature based on the previous cell
+    // and are thus no longer valid.
+    fe_values_inside.reset();
+    fe_values_outside.reset();
+
+    switch (current_face_location)
+      {
+        case LocationToLevelSet::inside:
+          {
+            call_reinit(*fe_values_inside_full_quadrature.at(active_fe_index));
+            break;
+          }
+        case LocationToLevelSet::outside:
+          {
+            call_reinit(*fe_values_outside_full_quadrature.at(active_fe_index));
+            break;
+          }
+        case LocationToLevelSet::intersected:
+          {
+            const unsigned int mapping_index =
+              mapping_collection->size() > 1 ? active_fe_index : 0;
+            const unsigned int q1D_index =
+              q_collection_1D.size() > 1 ? active_fe_index : 0;
+
+            quadrature_generator.set_1D_quadrature(q1D_index);
+            quadrature_generator.generate(cell, face_no);
+
+            const Quadrature<dim - 1> &inside_quadrature =
+              quadrature_generator.get_inside_quadrature();
+            const Quadrature<dim - 1> &outside_quadrature =
+              quadrature_generator.get_outside_quadrature();
```

Comment:
Could we amend these functions with a `face` pre/postfix? The latter two functions return different things how the `generate()` function was called. I think it would be better to be explicit.

Reply:
I don't understand this comment, can you elaborate a bit?

Reply:
```suggestion
            quadrature_generator.generate_face_quadrature(cell, face_no);

            const Quadrature<dim - 1> &inside_quadrature =
              quadrature_generator.get_inside_face_quadrature();
            const Quadrature<dim - 1> &outside_quadrature =
              quadrature_generator.get_outside_face_quadrature();
```

Reply:
But this comment comment doesn't relate to the class in this pull-request. Rather to how functions were named in these two classes:
https://www.dealii.org/developer/doxygen/deal.II/classNonMatching_1_1FaceQuadratureGenerator.html
https://www.dealii.org/developer/doxygen/deal.II/classNonMatching_1_1DiscreteFaceQuadratureGenerator.html

We could `quadrature_generator` to `face_quadrature_generator` maybe?

Reply:
OK. Let's do it as you have suggested!